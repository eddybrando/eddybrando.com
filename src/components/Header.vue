<template>
  <header class="header">
    <nav class="nav">
      <div>
        <TextLink
          v-for="({label, link}) in nav"
          class="link-block"
          :key="label"
          :label="label"
          :to="link"
        />
      </div>
    </nav>

    <transition name="fade" appear>
      <section class="title">
        <h1 class="headline mb-0">
          {{titlePrepend}}
          <span class="highlight" v-text="titleHighlight" />
          {{titleAppend}}
        </h1>
      </section>
    </transition>
  </header>
</template>

<script>
import TextLink from "~/components/TextLink";

export default {
  components: { TextLink },

  props: {
    nav: {
      required: true,
      type: Array
    },
    titleAppend: {
      required: true,
      type: String
    },
    titleHighlight: {
      required: true,
      type: String
    },
    titlePrepend: {
      required: true,
      type: String
    }
  }
};
</script>

<style lang="scss" scoped>
@import "~/assets/css/variables";

.fade-enter {
  opacity: 0;
}

.fade-enter-active {
  transition: opacity 0.75s;
}

.header {
  background-color: $color-blue;
  padding: 30px 40px 0;
  text-align: center;
}

.headline {
  font-size: 25px;
}

.highlight {
  background: linear-gradient(180deg, transparent 20%, $color-yellow 20%);
}

.link-block {
  margin-right: 40px;

  &:last-child {
    margin-right: 0;
  }
}

@media only screen and (min-width: $viewport-lg-min) {
  .header {
    display: grid;
    grid-template-rows: 1fr 200px 1fr;
    text-align: right;
  }

  .headline {
    font-size: 35px;
  }

  .link-block {
    font-size: 15px;
  }

  .nav {
    align-items: end;
    display: grid;
    justify-items: end;
  }

  .title {
    align-items: center;
    display: grid;
  }
}

@media only screen and (min-width: $viewport-xl-min) {
  .header {
    padding-left: 160px;
    padding-right: 80px;
  }
}
</style>
