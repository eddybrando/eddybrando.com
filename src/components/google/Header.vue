<template>
  <header>
    <div class>Search</div>
    <Nav class="nav" />
  </header>
</template>

<script>
import Nav from "~/components/google/Nav";

export default {
  components: { Nav },
};
</script>

<style lang="scss" scoped>
@import "~/assets/css/google/variables";

.nav {
  border-bottom: $border;
  padding-left: $main-margin-left-lg;
}
</style>
