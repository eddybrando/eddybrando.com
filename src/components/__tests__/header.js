import React from "react"
import renderer from "react-test-renderer"

import Header from "../header"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Header
          nav={[
            { label: "First link", link: "/" },
            { label: "Second link", link: "/second-link" },
          ]}
          titleAppend="dolor"
          titleHighlight="highlighted text"
          titlePrepend="Lorem ipsum"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
