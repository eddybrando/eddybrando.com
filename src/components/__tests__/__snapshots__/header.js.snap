// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <div
      className="header-links"
    >
      <a
        className="link-block"
        href="/"
        style={Object {}}
        title="First link"
      >
        First link
         
        <svg
          aria-hidden={true}
          className="link-arrow"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "willChange": "transform",
            }
          }
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.3 3.7L13.1 7.5 1 7.5 1 8.5 13.1 8.5 9.3 12.3 10 13 15 8 10 3z"
          />
        </svg>
      </a>
      <a
        className="link-block"
        href="/second-link"
        style={Object {}}
        title="Second link"
      >
        Second link
         
        <svg
          aria-hidden={true}
          className="link-arrow"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "willChange": "transform",
            }
          }
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.3 3.7L13.1 7.5 1 7.5 1 8.5 13.1 8.5 9.3 12.3 10 13 15 8 10 3z"
          />
        </svg>
      </a>
    </div>
  </nav>
  <section
    className="react-reveal title"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <h1
      className="headline mb-0"
    >
      Lorem ipsum
      <span
        className="highlight"
      >
        highlighted text
      </span>
      dolor
    </h1>
  </section>
</header>
`;
